src/app/components/forms/post/AppFormPost.vue

```diff
@@ -82,866 +86,881 @@ type FormPostModel = FiresidePost & {
 		}
-		return null;
 	}
 
-	async fetchTimezones() {
-		// Get timezones list.
-		this.timezones = await Timezone.getGroupedTimezones();
-		for (const region in this.timezones) {
-			for (const tz of this.timezones[region]) {
-				let offset = '';
-				if (tz.o > 0) {
-					offset = `+${tz.o / 3600}:00`;
-				} else if (tz.o < 0) {
-					offset = `-${-tz.o / 3600}:00`;
-				}
-				tz.label = `(UTC${offset}) ${tz.i}`;
-			}
-		}
+	if (files.length === 0) {
+		return;
 	}
 
-	async onPaste(e: ClipboardEvent) {
-		// Do not react to paste events when "Images" is not selected.
-		if (!!this.attachmentType && this.attachmentType !== FiresidePost.TYPE_MEDIA) {
-			return;
-		}
-		if (this.isUploadingPastedImage) {
-			return;
-		}
+	// Show image tray.
+	if (!enabledImages.value) {
+		enableImages();
+	}
 
-		// Validate clipboard data
-		if (!e.clipboardData) {
-			return;
-		}
+	// Upload
+	isUploadingPastedImage.value = true;
 
-		const items = e.clipboardData.items;
-		if (!items) {
-			return;
+	const $payload = await Api.sendRequest(
+		`/web/posts/manage/add-media/${form.formModel.id}`,
+		{},
+		{
+			file: files,
+			progress: e2 => (form.formModel._progress = e2),
+			noErrorRedirect: true,
 		}
+	);
 
-		const files = [];
+	isUploadingPastedImage.value = false;
 
-		for (let i = 0; i < items.length; i++) {
-			const item = items[i];
-			if (item.type.includes('image')) {
-				const blob = item.getAsFile();
-				if (blob) {
-					files.push(blob);
-				}
-			}
-		}
-
-		if (files.length === 0) {
-			return;
-		}
+	if ($payload.success) {
+		// Apply returned media items.
+		const mediaItems = MediaItem.populate($payload.mediaItems);
+		onMediaUploaded(mediaItems);
+	} else {
+		onMediaUploadFailed($payload.reason);
+	}
+}
 
-		// Show image tray.
-		if (!this.enabledImages) {
-			this.enableImages();
-		}
+function onVideoChanged(video: FiresidePostVideo | null) {
+	if (video === null) {
+		form.formModel.videos = [];
+	} else {
+		form.formModel.videos = [video];
+	}
+}
 
-		// Upload
-		this.isUploadingPastedImage = true;
+function onUploadingVideoStatusChanged(status: VideoStatus) {
+	uploadingVideoStatus.value = status;
+	emit('videoUploadStatusChange', uploadingVideoStatus.value);
+}
 
-		const $payload = await Api.sendRequest(
-			`/web/posts/manage/add-media/${this.formModel.id}`,
-			{},
-			{
-				file: files,
-				progress: e2 => this.setField('_progress', e2),
-				noErrorRedirect: true,
-			}
-		);
+function onVideoProviderChanged(provider: string) {
+	videoProvider.value = provider;
+}
 
-		this.isUploadingPastedImage = false;
+function onDisableVideoAttachment() {
+	disableAttachments();
+}
 
-		if ($payload.success) {
-			// Apply returned media items.
-			const mediaItems = MediaItem.populate($payload.mediaItems);
-			this.onMediaUploaded(mediaItems);
-		} else {
-			this.onMediaUploadFailed($payload.reason);
-		}
-	}
+function assignBackgroundId(backgroundId: number | null) {
+	form.formModel.background_id = backgroundId;
```

Comment:
We should probably only assign the backgroundId that we find from the background. This current code may potentially cause bugs.